src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -236,12 +236,21 @@ private IList<string> GetResourceNamesForCulture(CultureInfo culture)
             {
                 var names = new List<string>();
                 using (var cultureResourceStream = _resourceAssemblyWrapper.GetManifestResourceStream(resourceStreamName))
-                using (var resources = new ResourceReader(cultureResourceStream))
                 {
-                    foreach (DictionaryEntry entry in resources)
+                    if (cultureResourceStream != null)
                     {
-                        var resourceName = (string)entry.Key;
-                        names.Add(resourceName);
+                        using (var resources = new ResourceReader(cultureResourceStream))
+                        {
+                            foreach (DictionaryEntry entry in resources)
+                            {
+                                var resourceName = (string)entry.Key;
+                                names.Add(resourceName);
+                            }
+                        }
+                    }
+                    else
+                    {
+                        throw new MissingManifestResourceException();
```

Comment:
Should provide a message. The default one is `Unable to find manifest resource.` Would make it incredibly difficult for a user to decipher why something is failing.